CMakeLists.txt

```diff
@@ -186,18 +204,20 @@ set(UTILITY_HEADER_FILES core/utilities/Buffer.h
         core/utilities/LogUploader.h
         core/utilities/Service.h
         core/utilities/StringUtils.h
-        core/utilities/Timer.h
-        core/Version.h)
+        core/utilities/Timer.h)
 
 # path for interface files
 file(COPY core/utilities/ DESTINATION ${CMAKE_LIBRARY_INCLUDE_DIRECTORY}/core/utilities PATTERN *.cpp EXCLUDE)
-file(COPY core/Version.h DESTINATION ${CMAKE_LIBRARY_INCLUDE_DIRECTORY}/core/)
 
-add_library(${PROJECT_NAME}Utility SHARED ${UTILITY_SOURCE_FILES} ${UTILITY_HEADER_FILES})
+add_library(${PROJECT_NAME}Utility ${LIB_TYPE} ${UTILITY_SOURCE_FILES} ${UTILITY_HEADER_FILES})
 target_link_libraries(${PROJECT_NAME}Utility crypto)
+if (NOT ${BUILD_SHARED})
```

Comment:
I want to leave the option of static building just in case, potentially it might be more efficient for some projects where we might have more limited space?